### ISSUE TITLE
Make negotiateHeader match case insensitive

### ISSUE THREAD

**Issue by satyenr** [ubccr/kerby#1]
From `khttp/http.go`:

```go
var (
	negotiateHeader       = "Negotiate"
	wwwAuthenticateHeader = "WWW-Authenticate"
	authorizationHeader   = "Authorization"
)
```
```go
func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
        ......
	authReply := strings.Split(resp.Header.Get(wwwAuthenticateHeader), " ")
	if len(authReply) != 2 || authReply[0] != negotiateHeader {
		return nil, errors.New("khttp: server replied with invalid www-authenticate header")
	}
        ......
}
```
 I have seen services that return "negotiate" instead of "Negotiate" - causing Transport.RoundTrip() to return an error. The following patch fixes the problem:

```diff
diff --git a/khttp/http.go b/khttp/http.go
index de9ee986..db195d39 100644
--- a/khttp/http.go
+++ b/khttp/http.go
@@ -28,7 +28,7 @@ import (
 )
 
 var (
-       negotiateHeader       = "Negotiate"
+       negotiateHeader       = "negotiate"
        wwwAuthenticateHeader = "WWW-Authenticate"
        authorizationHeader   = "Authorization"
 )
@@ -83,7 +83,7 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
        }
 
        authReply := strings.Split(resp.Header.Get(wwwAuthenticateHeader), " ")
-       if len(authReply) != 2 || authReply[0] != negotiateHeader {
+       if len(authReply) != 2 || strings.ToLower(authReply[0]) != negotiateHeader {
                return nil, errors.New("khttp: server replied with invalid www-authenticate header")
        }
````

**Comment by satyenr** [comment]
I created a pull request (https://github.com/ubccr/kerby/pull/2) for this.

**Comment by aebruno** [comment]
Fixed in 28a111eb8f27d8067fc669d70d1eeb6b97d249c9
